### PR TITLE
Updated help templates and snfoundryup info

### LINKED ISSUE #1516: Add resources in help/installation output
### Which components does the task require to be changed?

snforge

### Description

_"Something like Fuel / Sway is doing when you install the toolchain, they display this message:"_
```
Read the Docs:

- Sway Book: <https://docs.fuel.network/docs/sway>
- Forc Book: <https://docs.fuel.network/docs/forc>
- Rust SDK Book: <https://docs.fuel.network/docs/fuels-rs>
- TypeScript SDK: <https://docs.fuel.network/docs/fuels-ts>

Join the Community:

- Follow us @SwayLang: <https://twitter.com/SwayLang>
- Ask questions on Discourse: <https://forum.fuel.network/>

Report Bugs:

- Sway Issues: <https://github.com/FuelLabs/sway/issues/new>

Sway Applications: <https://github.com/FuelLabs/sway-applications>
```

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**
Closes #1516 

## Introduced changes

- Adds resources list under long help and after foundry installation

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/forge/src/main.rs [hunk 1]
```diff
@@ -36,7 +36,36 @@ fn default_fuzzer_runs() -> NonZeroU32 {
...
+Use -h for short descriptions and --help for more details.
+
+{before-help}{usage-heading} {usage}
+
+{all-args}{after-help}
+",
+    after_help = "Read the docs: https://foundry-rs.github.io/starknet-foundry/",
+    after_long_help = "\
+Read the docs:
+- Foundry Book: https://foundry-rs.github.io/starknet-foundry/
```
Comment (drknzz) [inline crates/forge/src/main.rs:53]:
```suggestion
- Starknet Foundry Book: https://foundry-rs.github.io/starknet-foundry/
```

File: crates/forge/src/main.rs [hunk 2]
```diff
@@ -36,7 +36,36 @@ fn default_fuzzer_runs() -> NonZeroU32 {
...
+- Cairo Book: https://book.cairo-lang.org/
+- Starknet Book: https://book.starknet.io/
+- Starknet Documentation: https://docs.starknet.io/
+- Scarb Documentation: https://docs.swmansion.com/scarb/docs.html
+
+Join the community:
+- Follow core developers on X: https://twitter.com/swmansionxyz
+- Chat on Telegram: https://t.me/starknet_foundry
+- Get support on Telegram: https://t.me/starknet_foundry_support
```
Comment (drknzz) [inline crates/forge/src/main.rs:62]: Makes sense for the support group to be linked here, however maybe we could put it under some `Support` heading?

Would be more visible + probably less chance people go to the main Telegram group asking for support (as it's the first in order, above the support group)

File: crates/sncast/src/main.rs [hunk 3]
```diff
@@ -30,8 +30,36 @@ use tokio::runtime::Runtime;
...
+Use -h for short descriptions and --help for more details.
+
+{before-help}{usage-heading} {usage}
+
+{all-args}{after-help}
+",
+    after_help = "Read the docs: https://foundry-rs.github.io/starknet-foundry/",
+    after_long_help = "\
+Read the docs:
+- Foundry Book: https://foundry-rs.github.io/starknet-foundry/
```
Comment (THenry14) [inline crates/sncast/src/main.rs:47]:
```suggestion
- Starknet Foundry Book: https://foundry-rs.github.io/starknet-foundry/
```
same as above

File: scripts/snfoundryup [hunk 4]
```diff
@@ -32,6 +32,26 @@ Options:
   -c, --commit           Specify a specific commit hash to install from
 
 For more information, check out https://foundry-rs.github.io/starknet-foundry/getting-started/installation.html
+EOF
+}
+finished_installation() {
+  cat <<EOF
+
+Read the docs:
+- Foundry Book: https://foundry-rs.github.io/starknet-foundry/
```
Comment (THenry14) [inline scripts/snfoundryup:41]:
```suggestion
- Starknet Foundry Book: https://foundry-rs.github.io/starknet-foundry/
```

**Review by THenry14 — COMMENTED** [review]
do you mind updating this PR with rendered version(s) of -h/--help? It would be cool if we were able to easily assess what it will actually look like, after template expansions are done.

File: crates/forge/src/main.rs (hunk 2 above)
Comment (MaksymilianDemitraszek) [reply crates/forge/src/main.rs:62]: Would appreciate some suggestion here 🤔 
```
Engage on the contributors telegram channel:
Get support on the user support telegram channel:
```
wdyt?

File: scripts/snfoundryup [hunk 5]
```diff
@@ -32,6 +32,26 @@ Options:
   -c, --commit           Specify a specific commit hash to install from
 
 For more information, check out https://foundry-rs.github.io/starknet-foundry/getting-started/installation.html
+EOF
+}
+finished_installation() {
```
Comment (Arcticae) [inline scripts/snfoundryup:37]: this is just printing the info, let's name it after that

File: crates/forge/src/main.rs (hunk 2 above)
Comment (Arcticae) [reply crates/forge/src/main.rs:62]: Perhaps sth like this?
```
- Get support via Telegram: https://t.me/starknet_foundry_support
- Or discord: https://discord.gg/KZWaFtPZJf
- Or join our general chat (Telegram): https://t.me/starknet_foundry
```

**Review by Arcticae — APPROVED** [review]
LGTM, some nitpicks

File: crates/forge/src/main.rs (hunk 1 above)
Comment (MaksymilianDemitraszek) [reply crates/forge/src/main.rs:53]: Done

File: crates/forge/src/main.rs (hunk 2 above)
Comment (MaksymilianDemitraszek) [reply crates/forge/src/main.rs:62]: Done

File: crates/sncast/src/main.rs (hunk 3 above)
Comment (MaksymilianDemitraszek) [reply crates/sncast/src/main.rs:47]: Done

File: scripts/snfoundryup (hunk 4 above)
Comment (MaksymilianDemitraszek) [reply scripts/snfoundryup:41]: Done

File: scripts/snfoundryup (hunk 5 above)
Comment (MaksymilianDemitraszek) [reply scripts/snfoundryup:37]: renamed to `print_docs_and_community_info`

**Comment by MaksymilianDemitraszek** [comment]
Screenshot 2024-04-17 at 10 10 35
Screenshot 2024-04-17 at 10 10 51
@THenry14

**Pull request merged**